---
title: "big_tagseq_graphs"
author: "stevenguerin"
date: "2023-09-10"
output: html_document
---

Loading in packages and sorting the data 
```{r}
library(tidyverse)
library(DESeq2)
library(apeglm) #to install: BiocManager::install("apeglm")
#library(annotables) #to install: install.packages("devtools"); devtools::install_github("stephenturner/annotables")
library(biobroom) #to install: BiocManager::install("biobroom")
library(WGCNA) #to install: BiocManager::install("WGCNA") 
  #https://bioinformaticsworkbook.org/tutorials/wgcna.html#gsc.tab=0
  #https://alexslemonade.github.io/refinebio-examples/04-advanced-topics/network-analysis_rnaseq_01_wgcna.html
library(EnhancedVolcano) #BiocManager::install('EnhancedVolcano')
  #https://bioconductor.org/packages/devel/bioc/vignettes/EnhancedVolcano/inst/doc/EnhancedVolcano.html
library(org.Rn.eg.db) #to install: BiocManager::install("org.Rn.eg.db")
  #this is for annotation of the rat genes 
library(ggpubr)
library(pheatmap) #heatmap
library(stats) #for anovas
library(emmeans) #for anovas
library(CorLevelPlot) #devtools::install_github("kevinblighe/CorLevelPlot")
library(GO.db) #to install: BiocManager::install("GO.db") ;this is for go analysis 
library(AnnotationDbi) #to install: BiocManager::install("AnnotationDbi")
library(matrixStats)
library(ggthemes)

#formatting files 

setwd("/Users/stevenguerin/Documents/R Projects/tag-seq")

#exp design data
exp_design<- read.csv("~/Documents/R Projects/tag-seq/exp_design.csv")
#deleting the first column
colData<- exp_design[,2:ncol(exp_design)]
#Set row names to the gene names
rownames(colData) <- exp_design$id
#reordering data by row names
new_order1 <- sort(rownames(colData))
colData <- colData[new_order1, ]
#sorting colData data by region 
hip_colData <- colData%>%
  filter(region == "hippo")%>%
  subset(select = -region)

ec_colData <- colData%>%
  filter(region == "entor")%>%
  subset(select = -region)
  
#count data 
rawcounts<- read.csv("~/Documents/R Projects/tag-seq/big_count_data.csv", check.names= FALSE)
#deleting the first column
counts_data<- rawcounts[,2:ncol(rawcounts)]
#Set row names to the gene names
rownames(counts_data) <- rawcounts$gene_id
#Remove bottom 5 rows
counts_data <- counts_data[1:(nrow(counts_data)-5),]
#reordering data by column names
new_order2 <- sort(colnames(counts_data))
counts_data <- counts_data[, new_order2]
#sorting counts_data by region 
hip_counts_data<- counts_data[, grep("H", names(counts_data))]
ec_counts_data<- counts_data[, grep("E", names(counts_data))]

#Check to see if experimental design and count data frames match up, both should be TRUE 
#hippo data
all(colnames(hip_counts_data) %in% rownames(hip_colData))
all(colnames(hip_counts_data) == rownames(hip_colData))
#ec data
all(colnames(ec_counts_data) %in% rownames(ec_colData))
all(colnames(ec_counts_data) == rownames(ec_colData))
```

Hippocampus DESeq object 
```{r}
#hippocampus 
#construct DESeq object 
hip_dds<- DESeqDataSetFromMatrix(countData = hip_counts_data,
                       colData = hip_colData, 
                       design = ~treatment)
hip_dds

resultsNames(hip_dds)

#prefiltering, only keeping genes with more than XX reads 
hip_keep<- rowSums(counts(hip_dds)) >= 10
hip_dds<- hip_dds[hip_keep,]
hip_dds

#Run DESeq on object.size; use resultNames(hip_dds) to understand the different results we can look for in the dds object 
hip_dds<- DESeq(hip_dds)
hip_res <- results(hip_dds, contrast = c("treatment", "ethanol", "control")) #("condition", "level_to_compare", "base_level")

summary(hip_res)

#genes ranked by p-value
hip_resOrdered <- hip_res[order(hip_res$padj),]

#gene name annotation
hip_resOrdered_df <- as.data.frame(hip_resOrdered)
hip_resOrdered_df$symbol <- mapIds(org.Rn.eg.db, keys = rownames(hip_resOrdered_df), keytype = "ENSEMBL", column = "SYMBOL")
hip_resOrdered_df


#write.csv(hip_resOrdered_df, "~/Documents/R Projects/tag-seq/bigtagseq/hippocampus_big_deg.csv", row.names = TRUE)
```

Hippocampus DESeq object treatment + sex
```{r}
#hippocampus 
#construct DESeq object 
hip_dds_2<- DESeqDataSetFromMatrix(countData = hip_counts_data,
                       colData = hip_colData, 
                       design = ~treatment + sex)
hip_dds_2

resultsNames(hip_dds_2)

#prefiltering, only keeping genes with more than XX reads 
hip_keep_2<- rowSums(counts(hip_dds_2)) >= 10
hip_dds_2<- hip_dds_2[hip_keep_2,]
hip_dds_2

#Run DESeq on object.size; use resultNames(hip_dds) to understand the different results we can look for in the dds object 
hip_dds_2<- DESeq(hip_dds_2)
hip_res_2 <- results(hip_dds_2, contrast = c("treatment", "ethanol", "control")) #("condition", "level_to_compare", "base_level")

summary(hip_res_2)

#genes ranked by p-value
hip_resOrdered_2 <- hip_res_2[order(hip_res_2$pvalue),]

#gene name annotation
hip_resOrdered_df_2 <- as.data.frame(hip_resOrdered_2)
hip_resOrdered_df_2$symbol <- mapIds(org.Rn.eg.db, keys = rownames(hip_resOrdered_df_2), keytype = "ENSEMBL", column = "SYMBOL")
hip_resOrdered_df_2

hippo_DEGs <- hip_resOrdered_df_2 %>%
  filter(padj < 0.1) %>%
  rownames_to_column(var = "ensemblid")

#write.csv(hip_resOrdered_df, "~/Documents/R Projects/tag-seq/bigtagseq/hippocampus_big_deg.csv", row.names = TRUE)
```

DESeq object with interaction factor 
```{r}
#hippocampus 
#construct DESeq object 
hip_dds_3<- DESeqDataSetFromMatrix(countData = hip_counts_data,
                       colData = hip_colData, 
                       design = ~treatment + sex + treatment:sex)
hip_dds_3

resultsNames(hip_dds_3)

#prefiltering, only keeping genes with more than XX reads 
hip_keep_3<- rowSums(counts(hip_dds_3)) >= 10
hip_dds_3<- hip_dds_3[hip_keep_3,]
hip_dds_3

#Run DESeq on object.size; use resultNames(hip_dds) to understand the different results we can look for in the dds object 
hip_dds_3<- DESeq(hip_dds_3)
hip_res_3 <- results(hip_dds_3, contrast = c("treatment", "ethanol", "control")) #("condition", "level_to_compare", "base_level")

summary(hip_res_3)

#genes ranked by p-value
hip_resOrdered_3 <- hip_res_3[order(hip_res_3$pvalue),]

#filtering genes by log fold change 
hip_resOrdered_3 <- hip_resOrdered_3[abs(hip_resOrdered_3$log2FoldChange) > 0.2, ]
head(hip_resOrdered_3)

#gene name annotation
hip_resOrdered_df_3 <- as.data.frame(hip_resOrdered_3)
hip_resOrdered_df_3$symbol <- mapIds(org.Rn.eg.db, keys = rownames(hip_resOrdered_df_3), keytype = "ENSEMBL", column = "SYMBOL")

#saving DEGs filtered by p value as a csv file for analysis outside of r 
hippo_DEGs <- hip_resOrdered_df_3 %>%
  filter(pvalue < 0.1) %>%
  rownames_to_column(var = "ensemblid")

#write.csv(hippo_DEGs, "~/Documents/R Projects/tag-seq/bigtagseq/hippo_degs.csv", row.names = TRUE)
```

Hippocampus Males only DDS Object
```{r}
male_hip_counts_data <-  hip_counts_data[,c("1_H","10_H", "11_H", "12_H", "13_H", "2_H", "29_H", "30_H", "31_H", "34_H", "37_H", "38_H", "39_H", "4_H", "40_H", "5_H", "8_H", "9_H")]

#filtering for variance 
#quantile(male_hip_counts_data$rowvariance, probs = 0.75)
male_hip_counts_data <- male_hip_counts_data %>%
    mutate(rowvariance = rowVars(as.matrix(male_hip_counts_data))) %>%
    filter(rowvariance > 250) 

#removing the variance column   
male_hip_counts_data <- male_hip_counts_data[, -19]
    
male_hip_colData <- hip_colData%>%
  filter(sex == "male")

male_hip_dds<- DESeqDataSetFromMatrix(countData = male_hip_counts_data,
                       colData = male_hip_colData, 
                       design = ~ treatment)

#prefiltering, only keeping genes with more than XX reads 
male_hip_keep<- rowSums(counts(male_hip_dds)) >= 10
male_hip_dds<- male_hip_dds[male_hip_keep,]
male_hip_dds

#Run DESeq on object.size; use resultNames(hip_dds) to understand the different results we can look for in the dds object 
male_hip_dds<- DESeq(male_hip_dds)
male_hip_res <- results(male_hip_dds, contrast = c("treatment", "ethanol", "control")) #("condition", "level_to_compare", "base_level")

#genes ranked by pvalue
male_hip_resOrdered <- male_hip_res[order(male_hip_res$pvalue),]

#filtering genes by log fold change 
#male_hip_resOrdered <- male_hip_resOrdered[abs(male_hip_resOrdered$log2FoldChange) > 0.2, ]
#head(hip_resOrdered_3)

#gene name annotation
male_hip_resOrdered_df <- as.data.frame(male_hip_resOrdered)
male_hip_resOrdered_df$symbol <- mapIds(org.Rn.eg.db, keys = rownames(male_hip_resOrdered_df), keytype = "ENSEMBL", column = "SYMBOL")
male_hip_resOrdered_df

#saving DEGs filtered by p value as a csv file for analysis outside of r 
hippo_DEGs_male <- male_hip_resOrdered_df %>%
  filter(padj < 0.1) %>%
  rownames_to_column(var = "ensemblid")

hippo_DEGs_male 

#write.csv(hippo_DEGs_male$ensemblid, "male_hippo_DEGs.csv", row.names = FALSE)
```

plotting some genes from the males: 
```{r}
plotCounts(male_hip_dds, gene="ENSRNOG00000046834", intgroup="treatment")
```

Hippocampus Females only DDS Object
```{r}
fem_hip_counts_data <-  hip_counts_data[,c("16_H","17_H", "18_H", "19_H", "21_H", "22_H", "23_H", "24_H", "25_H", "28_H", "41_H", "43_H", "44_H", "45_H", "46_H", "47_H", "48_H", "49_H", "50_H", "52_H")]

fem_hip_colData <- hip_colData%>%
  filter(sex == "female")

fem_hip_dds<- DESeqDataSetFromMatrix(countData = fem_hip_counts_data,
                       colData = fem_hip_colData, 
                       design = ~ treatment)

#prefiltering, only keeping genes with more than XX reads 
fem_hip_keep<- rowSums(counts(fem_hip_dds)) >= 10
fem_hip_dds<- fem_hip_dds[fem_hip_keep,]

#Run DESeq on object.size; use resultNames(hip_dds) to understand the different results we can look for in the dds object 
fem_hip_dds<- DESeq(fem_hip_dds)
fem_hip_res <- results(fem_hip_dds, contrast = c("treatment", "ethanol", "control")) #("condition", "level_to_compare", "base_level")

#genes ranked by pvalue
fem_hip_resOrdered <- fem_hip_res[order(fem_hip_res$pvalue),]

#filtering genes by log fold change 
fem_hip_resOrdered <- fem_hip_resOrdered[abs(fem_hip_resOrdered$log2FoldChange) > 0.2, ]

#gene name annotation
fem_hip_resOrdered_df <- as.data.frame(fem_hip_resOrdered)
fem_hip_resOrdered_df$symbol <- mapIds(org.Rn.eg.db, keys = rownames(fem_hip_resOrdered_df), keytype = "ENSEMBL", column = "SYMBOL")
fem_hip_resOrdered_df

#saving DEGs filtered by p value as a csv file for analysis outside of r 
hippo_DEGs_female <- fem_hip_resOrdered_df %>%
  filter(padj < 0.05) %>%
  rownames_to_column(var = "ensemblid")

#write.csv(hippo_DEGs_female$ensemblid, "female_hippo_DEGs.csv", row.names = FALSE)
```

plotting some genes from the females: 
```{r}
plotCounts(hip_dds, gene="ENSRNOG00000001287", intgroup="treatment")
```

Entorhinal Cortex DSeq object 
```{r}
#entorhinal cortex 
#construct DESeq object 
ec_dds<- DESeqDataSetFromMatrix(countData = ec_counts_data,
                       colData = ec_colData, 
                       design = ~ treatment + sex + treatment:sex)
ec_dds

#prefiltering, only keeping genes with more than 10 reads 
ec_keep<- rowSums(counts(ec_dds)) >= 10
ec_dds<- ec_dds[ec_keep,]

#Run DESeq on object.size
ec_dds<- DESeq(ec_dds)
ec_res <- results(ec_dds, contrast = c("treatment", "ethanol", "control")) #("condition", "level_to_compare", "base_level")

#genes ranked by pvalue
ec_resOrdered <- ec_res[order(ec_res$pvalue),]

#gene name annotation
ec_resOrdered_df <- as.data.frame(ec_resOrdered)
ec_resOrdered_df$symbol <- mapIds(org.Rn.eg.db, keys = rownames(ec_resOrdered_df), keytype = "ENSEMBL", column = "SYMBOL")
ec_resOrdered_df

ec_DEGs <- ec_resOrdered_df %>%
  filter(pvalue < 0.05) %>%
  rownames_to_column(var = "ensemblid")

write.csv(ec_DEGs$ensemblid, "ec_DEGs.csv", row.names = FALSE)

#write.csv(ec_resOrdered_df, "~/Documents/R Projects/tag-seq/bigtagseq/entorhinal_big_deg.csv", row.names = TRUE)
```

Entorhinal Cortex Males only DDS Object
```{r}
male_ec_counts_data <-  ec_counts_data[,c("1_E","10_E", "11_E", "12_E", "13_E", "2_E", "29_E", "30_E", "31_E", "34_E", "35_E", "37_E", "38_E", "39_E", "4_E", "40_E", "5_E", "8_E", "9_E")]

male_ec_colData <- ec_colData%>%
  filter(sex == "male")

male_ec_dds<- DESeqDataSetFromMatrix(countData = male_ec_counts_data,
                       colData = male_ec_colData, 
                       design = ~ treatment)

#prefiltering, only keeping genes with more than XX reads 
male_ec_keep<- rowSums(counts(male_ec_dds)) >= 10
male_ec_dds<- male_ec_dds[male_ec_keep,]

#Run DESeq on object.size; use resultNames(ec_dds) to understand the different results we can look for in the dds object 
male_ec_dds<- DESeq(male_ec_dds)
male_ec_res <- results(male_ec_dds, contrast = c("treatment", "ethanol", "control")) #("condition", "level_to_compare", "base_level")

#genes ranked by pvalue
male_ec_resOrdered <- male_ec_res[order(male_ec_res$pvalue),]

#filtering genes by log fold change 
male_ec_resOrdered <- male_ec_resOrdered[abs(male_ec_resOrdered$log2FoldChange) > 0.2, ]

#gene name annotation
male_ec_resOrdered_df <- as.data.frame(male_ec_resOrdered)
male_ec_resOrdered_df$symbol <- mapIds(org.Rn.eg.db, keys = rownames(male_ec_resOrdered_df), keytype = "ENSEMBL", column = "SYMBOL")
male_ec_resOrdered_df

#saving DEGs filtered by p value as a csv file for analysis outside of r 
ec_DEGs_male <- male_ec_resOrdered_df %>%
  filter(pvalue < 0.05) %>%
  rownames_to_column(var = "ensemblid")

write.csv(ec_DEGs_male$ensemblid, "male_ec_DEGs.csv", row.names = FALSE)
```

Entorhinal Cortex Females only DDS Object
```{r}
fem_ec_counts_data <-  ec_counts_data[,c("16_E","17_E", "18_E", "19_E", "21_E", "22_E", "23_E", "24_E", "25_E", "28_E", "41_E", "43_E", "44_E", "45_E", "46_E", "47_E", "48_E", "49_E", "50_E", "52_E")]

fem_ec_colData <- ec_colData%>%
  filter(sex == "female")

fem_ec_dds<- DESeqDataSetFromMatrix(countData = fem_ec_counts_data,
                       colData = fem_ec_colData, 
                       design = ~ treatment)

#prefiltering, only keeping genes with more than XX reads 
fem_ec_keep<- rowSums(counts(fem_ec_dds)) >= 10
fem_ec_dds<- fem_ec_dds[fem_ec_keep,]

#Run DESeq on object.size; use resultNames(ec_dds) to understand the different results we can look for in the dds object 
fem_ec_dds<- DESeq(fem_ec_dds)
fem_ec_res <- results(fem_ec_dds, contrast = c("treatment", "ethanol", "control")) #("condition", "level_to_compare", "base_level")

#genes ranked by pvalue
fem_ec_resOrdered <- fem_ec_res[order(fem_ec_res$pvalue),]

#filtering genes by log fold change 
fem_ec_resOrdered <- fem_ec_resOrdered[abs(fem_ec_resOrdered$log2FoldChange) > 0.2, ]

#gene name annotation
fem_ec_resOrdered_df <- as.data.frame(fem_ec_resOrdered)
fem_ec_resOrdered_df$symbol <- mapIds(org.Rn.eg.db, keys = rownames(fem_ec_resOrdered_df), keytype = "ENSEMBL", column = "SYMBOL")
fem_ec_resOrdered_df

#saving DEGs filtered by p value as a csv file for analysis outside of r 
ec_DEGs_female <- fem_ec_resOrdered_df %>%
  filter(pvalue < 0.05) %>%
  rownames_to_column(var = "ensemblid")

write.csv(ec_DEGs_female$ensemblid, "female_ec_DEGs.csv", row.names = FALSE)
```

plotting some genes: 
```{r}
gfaphip <- plotCounts(hip_dds, gene="ENSRNOG00000002919", intgroup="treatment")

#plotCounts(ec_dds, gene="ENSRNOG00000002919", intgroup="treatment")

ggplot(gfaphip, aes(x=sex, y = count))+ 
  geom_jitter()+ 
  geom_boxplot(alpha = 0.5)+ 
  labs(title = "GFAP")+ 
  ylab("normalized count")
```

MA Plots
```{r}
#without shrinkage: 
plotMA(hip_res)

plotMA(ec_res)

#with shrinkage: 
#hippocampus
resultsNames(hip_dds)

hip_resLFC <- lfcShrink(hip_dds, coef="treatment_ethanol_vs_control", type="apeglm")
hip_resLFC

plotMA(hip_resLFC, ylim=c(-2,2))

#entorhinal cortex
resultsNames(ec_dds)

ec_resLFC <- lfcShrink(ec_dds, coef="treatment_ethanol_vs_control", type="apeglm")
ec_resLFC

plotMA(ec_resLFC, ylim=c(-2,2))
```

PCA Plots
```{r}
#hippocampus
hip_vsd <- vst(hip_dds)

plotPCA(hip_vsd, intgroup=c("treatment", "sex"))

#hippocampus
ec_vsd <- vst(ec_dds)

plotPCA(ec_vsd, intgroup=c("treatment", "sex"))
```

Volcano Plots
```{r}
#hippocampus model with ~treatment + sex + treatment:sex
EnhancedVolcano(hip_resOrdered_df_3,
  lab = hip_resOrdered_df_3$symbol,
  x = 'log2FoldChange',
  y = 'pvalue',
  boxedLabels = TRUE,
  drawConnectors = TRUE,
  pCutoff = 0.05, 
  FCcutoff = 0.1,
  xlim = c(-0.5, 0.5),
  ylim = c(0, 4.5),
  title = "Hippocampus- ~treatment + sex + treatment:sex",
  legendPosition = 'none',
  subtitle = "Differential Expresion"
)

#entorhinal cortex  
EnhancedVolcano(ec_resOrdered_df,
  lab = ec_resOrdered_df$symbol,
  x = 'log2FoldChange',
  y = 'pvalue', 
  boxedLabels = TRUE,
  drawConnectors = TRUE,
  pCutoff = 0.05, 
  FCcutoff = 0.1,
  xlim = c(-0.5, 0.5),
  ylim = c(0, 4.5),
  title = "Entorhinal Cortex",
  legendPosition = 'none',
  subtitle = "Differential Expresion"
)

```

Gene count graphs by region 
```{r}
#extracting the counts for genes of interest 
hip_genelist<- counts(hip_dds, normalized = FALSE)%>%
  as.data.frame()%>%
  tibble::rownames_to_column(var = "ENSEMBL")%>% #converting row name to a column 
  mutate(symbol = mapIds(org.Rn.eg.db, keys = ENSEMBL, keytype = "ENSEMBL", column = "SYMBOL"))%>%
  subset(select = -ENSEMBL)%>%
  filter(symbol %in% c("Clu","Gfap", "Apln", "Lxn", "Grin2c", "Aqp4", "Pltp", "Htra1", "Esr1", "Esr2"))

hip_genelist

#modyfing hip_colData so that the the rownames is a column 
hip_colData2 <- hip_colData%>%
  tibble::rownames_to_column(var = "id")

hip_colData2

#Gfap
#data frame
gfapgene<- hip_genelist%>%
  filter(symbol == "Gfap")%>%
  subset(select = -symbol)%>% #get rid of the symbol column
  t()%>% #transpose
  as.data.frame() %>%
  tibble::rownames_to_column(var = "id")%>%
  {colnames(.)[2] <- "count"; .}%>%
  full_join(hip_colData2, by = "id")

#anovas
gfapmodel <- lm(count ~ sex * treatment, data = gfapgene)
gfapanova <- anova(gfapmodel)
gfapanova

emmeans(gfapmodel, pairwise ~ sex + treatment)

#graph 
gfapcountsgraph <- ggboxplot(gfapgene, "sex", "count",
               color = "treatment", 
               add = "jitter",
               main = "Gfap Normalized Counts",
               palette = "jco") + 
    stat_compare_means(aes(group = sex), label = "p.signif",
                       label.y = 1500)

gfapcountsgraph

#Aqp4
#data frame
aqp4gene<- hip_genelist%>%
  filter(symbol == "Aqp4")%>%
  subset(select = -symbol)%>% #get rid of the symbol column
  t()%>% #transpose
  as.data.frame() %>%
  tibble::rownames_to_column(var = "id")%>%
  {colnames(.)[2] <- "count"; .}%>%
  full_join(hip_colData2, by = "id")

#anovas
aqp4model <- lm(count ~ sex * treatment, data = aqp4gene)
aqp4anova <- anova(aqp4model)
aqp4anova

emmeans(gfapmodel, pairwise ~ sex + treatment)

#graph 
aqp4countsgraph <- ggboxplot(aqp4gene, "sex", "count",
               color = "treatment", 
               add = "jitter",
               main = "Aqp4 Normalized Counts",
               palette = "jco") + 
    stat_compare_means(aes(group = sex), label = "p.signif",
                       label.y = 750)

aqp4countsgraph


ggarrange(gfapcountsgraph, aqp4countsgraph, ncol = 2, common.legend = TRUE)
```


GFAP gene counts for poster 
```{r}
#extracting the counts for genes of interest 
hip_genelist<- counts(hip_dds, normalized = FALSE)%>%
  as.data.frame()%>%
  tibble::rownames_to_column(var = "ENSEMBL")%>% #converting row name to a column 
  mutate(symbol = mapIds(org.Rn.eg.db, keys = ENSEMBL, keytype = "ENSEMBL", column = "SYMBOL"))%>%
  subset(select = -ENSEMBL)%>%
  filter(symbol %in% c("Clu","Gfap", "Apln", "Lxn", "Grin2c", "Aqp4", "Pltp", "Htra1", "Esr1", "Esr2"))

hip_genelist

#modyfing hip_colData so that the the rownames is a column 
hip_colData2 <- hip_colData%>%
  tibble::rownames_to_column(var = "id")

hip_colData2

#extracting the counts for genes of interest EC
ec_genelist<- counts(ec_dds, normalized = FALSE)%>%
  as.data.frame()%>%
  tibble::rownames_to_column(var = "ENSEMBL")%>% #converting row name to a column 
  mutate(symbol = mapIds(org.Rn.eg.db, keys = ENSEMBL, keytype = "ENSEMBL", column = "SYMBOL"))%>%
  subset(select = -ENSEMBL)%>%
  filter(symbol %in% c("Clu","Gfap", "Apln", "Lxn", "Grin2c", "Aqp4", "Pltp", "Htra1", "Esr1", "Esr2"))

ec_genelist

#modyfing hip_colData so that the the rownames is a column 
ec_colData2 <- ec_colData%>%
  tibble::rownames_to_column(var = "id")

ec_colData2

#Gfap Hip
#data frame
gfapgene<- hip_genelist%>%
  filter(symbol == "Gfap")%>%
  subset(select = -symbol)%>% #get rid of the symbol column
  t()%>% #transpose
  as.data.frame() %>%
  tibble::rownames_to_column(var = "id")%>%
  {colnames(.)[2] <- "count"; .}%>%
  full_join(hip_colData2, by = "id")

#anovas
gfapmodel <- lm(count ~ sex * treatment, data = gfapgene)
gfapanova <- anova(gfapmodel)
gfapanova

emmeans(gfapmodel, pairwise ~ sex + treatment)

#graph 
gfapcountsgraph <- ggplot(gfapgene, aes(x = sex, y = count, color = treatment)) + 
  geom_boxplot() + 
  geom_jitter(position = position_jitterdodge()) + 
  scale_fill_manual(values = c("#00AFBB", "#E7B800"))+
  scale_color_manual(values = c("#00AFBB", "#E7B800"))+
  theme_minimal() 

gfapcountsgraph

#Gfap EC
#data frame
gfapgeneec<- ec_genelist%>%
  filter(symbol == "Gfap")%>%
  subset(select = -symbol)%>% #get rid of the symbol column
  t()%>% #transpose
  as.data.frame() %>%
  tibble::rownames_to_column(var = "id")%>%
  {colnames(.)[2] <- "count"; .}%>%
  full_join(ec_colData2, by = "id")

#anovas
gfapmodelec <- lm(count ~ sex * treatment, data = gfapgeneec)
gfapanovaec <- anova(gfapmodelec)
gfapanovaec

emmeans(gfapmodelec, pairwise ~ sex + treatment)

#graph 
gfapcountsgraphec <- ggplot(gfapgeneec, aes(x = sex, y = count, color = treatment)) + 
  geom_boxplot() + 
  geom_jitter(position = position_jitterdodge()) + 
  scale_fill_manual(values = c("#00AFBB", "#E7B800"))+
  scale_color_manual(values = c("#00AFBB", "#E7B800"))+
  theme_minimal()  

gfapcountsgraphec


ggarrange(gfapcountsgraph, gfapcountsgraphec, ncol = 2, common.legend = TRUE)
```
Heatmaps
```{r}
#Hippocampus
#list of genes we want to use
hip_genelist<- counts(hip_dds, normalized = TRUE)%>%
  as.data.frame()%>%
  tibble::rownames_to_column(var = "ENSEMBL")%>% #converting row name to a column 
  mutate(symbol = mapIds(org.Rn.eg.db, keys = ENSEMBL, keytype = "ENSEMBL", column = "SYMBOL"))%>%
  subset(select = -ENSEMBL)%>%
  filter(symbol %in% c("Clu","Gfap", "Apln", "Lxn", "Grin2c", "Aqp4", "Pltp", "Htra1"))

#taking the df you made in the previous chunk and editing it a bit 
row.names(hip_genelist) <- hip_genelist$symbol
hip_genelist<- hip_genelist%>%
  subset(select = -symbol)

#arranging hip_colData
hip_colData_arranged <- hip_colData%>%
  arrange(treatment,sex)

hip_colData_order <- rownames(hip_colData_arranged)
hip_genelist_reordered <- hip_genelist[, hip_colData_order]

#heatmap
hip_heatmap<- pheatmap(
  hip_genelist_reordered,
  cluster_rows = FALSE,
  cluster_cols = FALSE, 
  annotation= hip_colData,
  show_rownames = TRUE,
  gaps_col = c(10, 18, 28),
  gaps_row = 7,
  border_color = FALSE,
  main = "Normalized Counts of DEGs in the Hippocampus",
  scale = "row"
)

#Entorhinal Cortex
#list of genes we want to use
ec_genelist<- counts(ec_dds, normalized = TRUE)%>%
  as.data.frame()%>%
  tibble::rownames_to_column(var = "ENSEMBL")%>% #converting row name to a column 
  mutate(symbol = mapIds(org.Rn.eg.db, keys = ENSEMBL, keytype = "ENSEMBL", column = "SYMBOL"))%>%
  subset(select = -ENSEMBL)%>%
  filter(symbol %in% c("Gfap","Aqp4"))

#taking the df you made in the previous chunk and editing it a bit 
row.names(ec_genelist) <- ec_genelist$symbol
ec_genelist<- ec_genelist%>%
  subset(select = -symbol)

#arranging hip_colData
ec_colData_arranged <- ec_colData%>%
  arrange(treatment,sex)

ec_colData_order <- rownames(ec_colData_arranged)
ec_genelist_reordered <- ec_genelist[, ec_colData_order]

#heatmap
ec_heatmap<- pheatmap(
  ec_genelist_reordered,
  cluster_rows = FALSE,
  cluster_cols = FALSE, 
  annotation= ec_colData,
  show_rownames = TRUE,
  gaps_col = c(10, 19, 29),
  border_color = FALSE,
  cellheight = 25,
  main = "Normalized Counts of DEGs in the Entorhinal Cortex",
  scale = "row"
)
```

WGCNA
```{r}
##Tutorials:
#https://rstudio-pubs-static.s3.amazonaws.com/687551_ed469310d8ea4652991a2e850b0018de.html
#https://www.youtube.com/watch?v=gYE59uEMXT4&t=419s
#https://wikis.utexas.edu/display/bioiteam/Clustering+using+WGCNA

#detect outlier genes
gsg<- goodSamplesGenes(t(hip_counts_data))
summary(gsg)
gsg$allOK #result is false- so there are outliers

table(gsg$goodGenes)
table(gsg$goodSamples)

#remove genes that are detected as outliers
data_nooutlier <- hip_counts_data[gsg$goodGenes == TRUE,]

#remove outlier samples - hierarchical clustering (method 1)
htree <- hclust(dist(t(data_nooutlier)), method = "average")
plot(htree)

#pca (method 2 for identifying outliers)
pca <- prcomp(t(hip_counts_data))
pca.dat <- pca$x

pca.var <- pca$sdev^2
pca.var.percent <- round(pca.var/sum(pca.var) *100, digits = 2)

pca.dat <- as.data.frame(pca.dat)

ggplot(pca.dat, aes(PC1, PC2)) + 
  geom_point() + 
  geom_text(label = rownames(pca.dat)) + 
  labs(x = paste0('PC1: ',pca.var.percent[1], ' %'),
       y = paste0('PC2: ', pca.var.percent[2], ' %'))


# Normalization with DESeq
#ensuring ronames and column names are identical
all(rownames(hip_colData) %in% colnames(hip_counts_data))
all(rownames(hip_colData) == colnames(hip_counts_data))

#create dds 
hip_dds2 <- DESeqDataSetFromMatrix(countData = hip_counts_data, 
                                   colData = hip_colData, 
                                   design = ~1) #not specifying a model 

#remove all genes with counts > 15 in more than 75% of samples
hip_dds_75 <- hip_dds2[rowSums(counts(hip_dds2) >= 15) >= 24,]
nrow(hip_dds_75) #9809 genes left 

#perform variance stabilization
hip_dds_norm <- vst(hip_dds_75)

#get normalized counts
hip_norm.counts <- assay(hip_dds_norm) %>%
  t()

# Network Construction --------
power <- c(c(1:10), seq(from = 12, to = 50, by = 2))

#call the network topology analysis function 
hip_sft <- pickSoftThreshold(hip_norm.counts, 
                  powerVector = power, 
                  networkType = "signed",
                  verbose = 5)

hip_sft.data <- hip_sft$fitIndices

#visualization to pick power
hip_a1 <- ggplot(hip_sft.data, aes(Power, SFT.R.sq, label = Power)) + 
  geom_point() + 
  geom_text(nudge_y = 0.1) + 
  geom_hline(yintercept = 0.8, color = 'red') + 
  labs(x= 'Power', y = 'Scale free topology model fit, signed R^2') + 
  theme_classic()

hip_a1

hip_a2 <- ggplot(hip_sft.data, aes(Power, mean.k., label= Power)) + 
  geom_point() + 
  geom_text(nudge_y = 0.1) + 
  labs(x= 'Power', y = 'Mean Connectivity') + 
  theme_classic()

hip_a2

# convert matrix to numeric 

hip_norm.counts[] <- sapply(hip_norm.counts, as.numeric)

soft_power <- 3
temp_cor <- cor
cor <- WGCNA::cor

#memory estimate w.r.t. blocksisze
bwnet <- blockwiseModules(hip_norm.counts, 
                 maxBlockSize = 14000,
                 TOMType = "signed", 
                 power = soft_power, 
                 mergeCutHeight = 0.25, 
                 numericLabels = FALSE,
                 randomSeed = 1234,
                 verbose = 3)

cor <- temp_cor

# Moduel Eigengenes

module_eigengenes <- bwnet$MEs

#Print out a preview
head(module_eigengenes)

#get number of genes for each module
table(bwnet$colors)

#plot a dendogram and the module colors before and after merging underneath
plotDendroAndColors(bwnet$dendrograms[[1]], cbind(bwnet$unmergedColors, bwnet$colors),
                    c("unmerged", "merged"),
                    dendroLabels = FALSE,
                    addGuide = TRUE, 
                    hang = 0.03, 
                    guideHang = 0.05)

#create a traits file- binarize categorical variables 
hip_colData_mut <- hip_colData %>%
  mutate(treatment_state_bin = ifelse(grepl('ethanol', treatment), 1, 0)) %>%
  mutate(sex_state_bin = ifelse(grepl('female', sex), 1, 0)) %>%
  dplyr::select(3,4)
  
# define numbers of genes and samples
nSamples <- nrow(hip_norm.counts)
nGenes <- ncol(hip_norm.counts)

module.trait.corr <- cor(module_eigengenes, hip_colData_mut, use = 'p')
module.trait.corr.pvals <- corPvalueStudent(module.trait.corr, nSamples)

#visualize module-trait association as a heatmap
heatmap.data <- merge(module_eigengenes, hip_colData_mut, by = 'row.names')

head(heatmap.data)

hip_colData_mut <- hip_colData_mut %>%
  column_to_rownames(var = 'id') #this might be wrong 

CorLevelPlot(heatmap.data,
             x= names(heatmap.data)[24:25],
             y= names(heatmap.data)[2:23],
             col = c("blue1", "skyblue", "white", "pink", "red"))

module.gene.mapping <- as.data.frame(bwnet$colors)
grey_module <- module.gene.mapping %>%
  filter(`bwnet$colors` == "grey") #filter by color for a specific module 
  

#grey, lightyellow, and red are all associated with ethanol treatment 
```

GO analysis of WGCNA modules 
```{r}
#tutorials
#https://lauren-blake.github.io/Explore_WGCNA/Tutorial_1.html

GOenr = GOenrichmentAnalysis(grey_module, allLLIDs, organism = "rat", nBestP = 10);



```

